iree/compiler/InputConversion/MHLO/MHLOToLinalgOnTensors.cpp

```diff
@@ -268,6 +268,101 @@ struct ConstOpConversion : public OpConversionPattern<mhlo::ConstOp> {
   }
 };
 
+//===----------------------------------------------------------------------===//
+// mhlo.RngUniformOp conversion patterns.
+//===----------------------------------------------------------------------===//
+
+namespace {
+/// Returns an ArrayAttr that contains `nLoops` attributes. All the attributes
+/// are "parallel" except the last `nReduction` elements, where are "reduction"
+/// attributes.
+SmallVector<StringRef, 3> GetParallelAndReductionIterators(
+    unsigned nLoops, unsigned nReduction) {
+  SmallVector<StringRef, 3> res(nLoops - nReduction,
+                                getParallelIteratorTypeName());
+  res.append(nReduction, getReductionIteratorTypeName());
+  return res;
+}
+
+struct RngUniformConversion : public OpConversionPattern<mhlo::RngUniformOp> {
+  using OpConversionPattern<mhlo::RngUniformOp>::OpConversionPattern;
+
+  LogicalResult matchAndRewrite(
+      mhlo::RngUniformOp op, ArrayRef<Value> args,
+      ConversionPatternRewriter &rewriter) const final {
+    Type int32Type = IntegerType::get(op.getContext(), /*width=*/32);
+    auto loc = op.getLoc();
+    auto targetTy = op->getResult(0).getType().dyn_cast<ShapedType>();
+    auto targetRank = targetTy.getRank();
+    Value output;
+    Value initTensor = rewriter.create<linalg::InitTensorOp>(
+        loc, ArrayRef<Value>({}), targetTy.getShape(),
+        targetTy.getElementType());
+    auto resultTy = this->typeConverter->convertType(op.getResult().getType())
+                        .cast<ShapedType>();
+    // Creates index map using target matrix's rank.
+    SmallVector<AffineMap, 3> indexingMaps;
+    indexingMaps.push_back(AffineMap::get(targetRank, /*symbolCount=*/0,
+                                          SmallVector<AffineExpr>({}),
+                                          rewriter.getContext()));
+    indexingMaps.push_back(AffineMap::get(targetRank, /*symbolCount=*/0,
+                                          SmallVector<AffineExpr>({}),
+                                          rewriter.getContext()));
+    SmallVector<AffineExpr> outputExprs;
+    for (int i = 0; i < targetRank; i++) {
+      outputExprs.push_back(rewriter.getAffineDimExpr(i));
+    }
+    indexingMaps.push_back(AffineMap::get(targetRank, /*symbolCount=*/0,
+                                          outputExprs, rewriter.getContext()));
+    // Generic region with LCG Algorithm that make use of element index from:
+    // https://reviews.llvm.org/D101364
+    auto linalgOp = rewriter.create<linalg::GenericOp>(
+        op.getLoc(), /*resultTensors=*/resultTy,
+        /*inputs=*/ValueRange{args[0], args[1]},
+        /*outputs=*/initTensor, indexingMaps,
+        GetParallelAndReductionIterators(/*nLoops=*/targetRank,
+                                         /*nReduction=*/0),
+        [&](OpBuilder &b, Location loc, ValueRange args) {
+          llvm::SmallVector<Value> updateVec = {
+              b.create<ConstantOp>(loc, b.getI32IntegerAttr(0))};
```

Comment:
Is it the initial seed? I think it's worth to have a var for the seed number. It's more clear to me.

Reply:
done